### ISSUE TITLE
Ledger Nano X automatically allows requests when create wallet or verify address

### ISSUE THREAD

**Issue by yanguoyu** [Magickbase/neuron-public-issues#226]
Refer to https://github.com/nervosnetwork/neuron/issues/2749

**Comment by yanguoyu** [comment]
https://github.com/LedgerHQ/app-nervos/pull/22/commits/bf03d1caa9586b262994e2c4110cc3cfe1451ddd add a flag to check whether `app-nervos` should show the verify dialog. If the `p1` is set as `0x1`, it will show verify, otherwise, it will not.

But https://github.com/obsidiansystems/hw-app-ckb/blob/develop/src/Ckb.js#L50 and https://github.com/obsidiansystems/hw-app-ckb/blob/develop/src/Ckb.js#L103 all set `p1` as `0x0`, so the ledger will not show verify dialog after `app-nervos` update to 0.5.6 when using `Neuron`.

https://github.com/obsidiansystems/hw-app-ckb/ does not seem to have continued maintenance. So we have two ways to resolve this issue.
1. Copy the `getWalletExtendedPublicKey` and `getWalletPublicKey` to our codes, and change `0x0` to `0x1`
2. Fork the repo and update the code and then release it with npm.

The first solution is easy and quick, second is a little complex. I would suggest the first.
What do you think?  @Keith-CY @homura

**Comment by Keith-CY** [comment]
> [LedgerHQ/app-nervos@bf03d1c](https://github.com/LedgerHQ/app-nervos/commit/bf03d1caa9586b262994e2c4110cc3cfe1451ddd) add a flag to check whether `app-nervos` should show the verify dialog. If the `p1` is set as `0x1`, it will show verify, otherwise, it will not.
> 
> But [obsidiansystems/hw-app-ckb@`develop`/src/Ckb.js#L50](https://github.com/obsidiansystems/hw-app-ckb/blob/develop/src/Ckb.js?rgh-link-date=2023-07-21T03%3A25%3A50Z#L50) and [obsidiansystems/hw-app-ckb@`develop`/src/Ckb.js#L103](https://github.com/obsidiansystems/hw-app-ckb/blob/develop/src/Ckb.js?rgh-link-date=2023-07-21T03%3A25%3A50Z#L103) all set `p1` as `0x0`, so the ledger will not show verify dialog after `app-nervos` update to 0.5.6 when using `Neuron`.
> 
> [obsidiansystems/hw-app-ckb](https://github.com/obsidiansystems/hw-app-ckb/) does not seem to have continued maintenance. So we have two ways to resolve this issue.
> 
> 1. Copy the `getWalletExtendedPublicKey` and `getWalletPublicKey` to our codes, and change `0x0` to `0x1`
> 2. Fork the repo and update the code and then release it with npm.
> 
> The first solution is easy and quick, second is a little complex. I would suggest the first. What do you think? @Keith-CY @homura

Based on the analysis above, we can see that

`p1` is set `0x00` as default value at https://github.com/obsidiansystems/hw-app-ckb/blob/develop/src/Ckb.js#L49

It is loaded as `verify` at https://github.com/LedgerHQ/app-nervos/commit/bf03d1caa9586b262994e2c4110cc3cfe1451ddd#diff-433413ac838af0a855dd6763c79cb3b82e5b44a18230b19842b7d2d4c060f98aR115

So the branch to prompt is skipped by default at https://github.com/LedgerHQ/app-nervos/commit/bf03d1caa9586b262994e2c4110cc3cfe1451ddd#diff-433413ac838af0a855dd6763c79cb3b82e5b44a18230b19842b7d2d4c060f98aR130

It's not a good pattern to skip user confirmation by default, and an update should be pushed to https://github.com/LedgerHQ/app-nervos

The last commit on https://github.com/LedgerHQ/app-nervos was posted on May. 31, so it's still maintained, I'll try to connect to the ledger team for this issue

**Comment by homura** [comment]
I am confused about which of https://github.com/LedgerHQ/app-nervos and https://github.com/obsidiansystems/ledger-app-nervos is the "official" app, or which one is the most installed app. The "verify" feature seems only supported in the [LedgerHQ](https://github.com/LedgerHQ/app-nervos/commit/bf03d1caa9586b262994e2c4110cc3cfe1451ddd)

I think we need to check the following

- Did Ledger installed a "nervos-app" by default? 
  - if so, which one did Ledger installed by default
- Is there a mix of both apps? 
  - if so, we need to determine in Neuron which app the user installed in the hardware wallet to decide which API to call

**Comment by homura** [comment]
> * Did Ledger installed a "nervos-app" by default?
>   
>   * if so, which one did Ledger installed by default

No, and I've got a response from @yanguoyu that Ledger offers an app market, and the marketplace offers an app based on https://github.com/LedgerHQ/app-nervos 

> * Is there a mix of both apps?
>   * if so, we need to determine in Neuron which app the user installed in the hardware wallet to decide which API to call

Probably not, unless the user is installing from a GitHub release and not from the Ledger marketplace.

So I'd vote for `2. Fork the repo and update the code and then release it with npm.`, the reason is 

1. Neuron have been running with `hw-app-ckb` for a long time, which is the equivalent of verifying the stability and correctness of the SDK
2. the fewer changes mean fewer risks

**Comment by yanguoyu** [comment]
> It's not a good pattern to skip user confirmation by default, and an update should be pushed to https://github.com/LedgerHQ/app-nervos

I agree with that, and the app update should not change the default behavior. 

> the fewer changes mean fewer risks
Actually, the first solution also uses the `hw-app-ckb` code and only changes the default `p1` value. The difference is we will copy codes from `hw-app-ckb` to `neuron.`

**Comment by Keith-CY** [comment]
> > * Did Ledger installed a "nervos-app" by default?
> >   
> >   * if so, which one did Ledger installed by default
> 
> No, and I've got a response from @yanguoyu that Ledger offers an app market, and the marketplace offers an app based on [LedgerHQ/app-nervos](https://github.com/LedgerHQ/app-nervos)
> 
> > * Is there a mix of both apps?
> >   
> >   * if so, we need to determine in Neuron which app the user installed in the hardware wallet to decide which API to call
> 
> Probably not, unless the user is installing from a GitHub release and not from the Ledger marketplace, the use case is too rare
> 
> So I'd vote for `2. Fork the repo and update the code and then release it with npm.`, the reason is
> 
> 1. Neuron have been running with `hw-app-ckb` for a long time, which shows stability and correctness of the SDK
> 2. the fewer changes mean fewer risks

I've opened a ticket in ledger discord and seek support from them.

Fork the repo to maintain another one is out of our ability because it should work with the one on their live app market, and we cannot detect a new release timely. The workflow to release `app-nervos` is obscure, only tags can be found at https://github.com/LedgerHQ/app-nervos/tags

**Comment by yanguoyu** [comment]
> Forking the repo to maintain another one

We can fork https://github.com/obsidiansystems/hw-app-ckb to fix it, but not need to fork the ledger app repo https://github.com/obsidiansystems/ledger-app-nervos.

**Comment by Keith-CY** [comment]
Ledger team has taken over this issue

**Comment by Keith-CY** [comment]
None response from the ledger team yet
